packages/chat-ui/src/components/CreateChat/wizardSubmit.ts

```diff
@@ -100,255 +100,249 @@ const getJsonFile = (arg0: {}) =>
-              curve: curveOut!.output.curve,
-            };
-
-            const metaOut = await chatSdk.createMetadataForBondingInstructions({
-              targetMintKeypair,
-              metadataUpdateAuthority: chatSdk.wallet.publicKey,
-              metadata,
-              decimals: bondingOpts.targetMintDecimals,
-            });
-
-            const bondingOut =
-              await tokenBondingSdk.createTokenBondingInstructions(bondingOpts);
-
-            postPermissionKey = bondingOut.output.targetMint;
-            postPermissionInstructions.push(
-              [...curveOut!.instructions!, ...metaOut!.instructions],
-              bondingOut!.instructions
-            );
-
-            postPermissionSigners.push(
-              [...curveOut!.signers, ...metaOut!.signers],
-              bondingOut!.signers
-            );
-
-            postPermissionKey = targetMintKeypair.publicKey;
-          } catch (e) {
-            innerError = e as Error;
-            setState({
-              error: e as Error,
-            });
-          }
-        }
-      }
-    }
-
-    if (!innerError) {
-      const chatOut = await chatSdk.initializeChatInstructions({
-        name,
-        identifier,
-        imageUrl,
-        permissions: {
-          readPermissionKey: readPermissionKey!,
-          defaultReadPermissionAmount: readPermissionAmount,
-          readPermissionType,
-          postPermissionKey: postPermissionKey!,
-          postPermissionAmount,
-          postPermissionType,
-        },
+  if (!chatSdk && !tokenBondingSdk) return
+
+  setState({ status: "submitting", error: undefined });
+  let readPermissionType, readPermissionAmount, readPermissionKey;
+  let postPermissionType, postPermissionAmount, postPermissionKey;
+  let readPermissionInstructions: TransactionInstruction[][] = [],
+    readPermissionSigners: Signer[][] = [],
+    postPermissionInstructions: TransactionInstruction[][] = [],
+    postPermissionSigners: Signer[][] = [];
+
+  readPermissionType = getPermissionType(readForm.type);
+  readPermissionAmount = readForm.amount;
+  readPermissionKey =
+    readForm.type === "native" || readForm.type === "token"
+      ? readForm.mint
+        ? new PublicKey(readForm.mint)
+        : undefined
+      : readForm.type === "nft"
+      ? new PublicKey(readForm.collectionKey!)
+      : undefined;
+
+  postPermissionType = getPermissionType(postForm.type);
+  postPermissionAmount = postForm.amount;
+  postPermissionKey =
+    postForm.type === "native" || postForm.type === "token"
+      ? postForm.mint
+        ? new PublicKey(postForm.mint)
+        : undefined
+      : postForm.type === "nft"
+      ? new PublicKey(postForm.collectionKey!)
+      : undefined;
+
+  let chatMetadataUri = "";
+  // Upload chat metadata json
+  if (wizardData.description && wizardData.description != "") {
+    try {
+      setState({
+        subStatus: "Uploading read token metadata to SHDW drive...",
```

Comment:
Nice job ripping this logic out. But this should flip between `Uploading read` or `Uploading post` depending on which token its creating.

Reply:
Thanks, fixed

Reply:
Thanks, fixed